driver-core/src/main/com/mongodb/internal/connection/tlschannel/async/AsynchronousTlsChannelGroup.java

```diff
@@ -383,7 +384,7 @@ private void waitForSocketRegistration(RegisteredSocket socket) {
         try {
             socket.registered.await();
         } catch (InterruptedException e) {
-            throw new RuntimeException(e);
+            throw interruptAndCreateMongoInterruptedException(null, e);
```

Comment:
This is a change in the code we vendor. As far as I know, we don't have a process for tracking such changes.

Reply:
The thing is that we'll never be able to make it the way we need: the driver is supposed to wrap `InterruptedException`s in `MongoInterruptedException`, but it won't do so when `InterruptedException` is thrown from this code.

I'll create a PR agains the `tlschannel`. If you check the history of changes in this file, you'll see we made driver-specific changes, that are not part of the original library. Given this, do you still want the change to be rolled back, and us not to wrap `InterruptedException` in `MongoInterruptedException` here?

Reply:
I guess it's ok to leave it as is.  It's just a fragile situation we're in here, since we don't have a process for re-vendoring.

Reply:
Created https://github.com/marianobarrios/tls-channel/pull/165.